homeassistant/components/vulcan/config_flow.py

```diff
@@ -0,0 +1,342 @@
+            data_schema=vol.Schema(data_schema),
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Reauthorize integration."""
+        errors = {}
+        if user_input is not None:
+            try:
+                credentials = await register(
+                    self.hass,
+                    user_input[CONF_TOKEN],
+                    user_input[CONF_REGION],
+                    user_input[CONF_PIN],
+                )
+            except VulcanAPIException as err:
+                if str(err) == "Invalid token!" or str(err) == "Invalid token.":
+                    errors["base"] = "invalid_token"
+                elif str(err) == "Expired token.":
+                    errors["base"] = "expired_token"
+                elif str(err) == "Invalid PIN.":
+                    errors["base"] = "invalid_pin"
+                else:
+                    errors["base"] = "unknown"
+                    _LOGGER.error(err)
+            except RuntimeError as err:
+                if str(err) == "Internal Server Error (ArgumentException)":
+                    errors["base"] = "invalid_symbol"
+                else:
+                    errors["base"] = "unknown"
+                    _LOGGER.error(err)
+            except ClientConnectionError as err:
+                errors["base"] = "cannot_connect"
+                _LOGGER.error("Connection error: %s", err)
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            if not errors:
+                account = credentials["account"]
+                keystore = credentials["keystore"]
+                client = Vulcan(keystore, account)
+                students = await client.get_students()
+                await client.close()
+                existing_entries = []
+                for entry in self.hass.config_entries.async_entries(DOMAIN):
+                    existing_entries.append(entry)
+                for student in students:
+                    for entry in existing_entries:
+                        if str(student.pupil.id) == str(entry.data["student_id"]):
+                            self.hass.config_entries.async_update_entry(
+                                entry,
+                                title=f"{student.pupil.first_name} {student.pupil.last_name}",
+                                data={
+                                    "student_id": str(student.pupil.id),
+                                    "keystore": keystore.as_dict,
+                                    "account": account.as_dict,
+                                },
+                            )
+                            await self.hass.config_entries.async_reload(entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth",
```

Comment:
There should be a separate confirm step that is used in the reauth flow. The input to the reauth step will always be the existing config entry data. The user input should come in the confirm step. Please see our docs and existing examples among integrations.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#reauthentication